netmiko/audiocode/audiocode_ssh.py

```diff
@@ -1,363 +1,735 @@
+		:type delay_factor: int
+
+		:param pattern: Regular expression pattern to determine whether prompt is valid
+
+		:param confirm_pattern: Regular expression pattern to confirm prompt was found by auto discovery.
+		"""
+		return super().find_prompt(
+			delay_factor=delay_factor, 
+			pattern=pattern,
+		)
+
+	def check_config_mode(
+		self, 
+		check_string: str = r"(\)#|\)\*#)", 
+		pattern: str = "#", 
+		force_regex: bool = True
+		) -> bool:
+		"""Checks if the device is in configuration mode or not.
+
+		:param check_string: Identification of configuration mode from the device
+		:type check_string: str
+
+		:param pattern: Pattern to terminate reading of channel
+		:type pattern: str
+		"""
+		return super().check_config_mode(
+			check_string=check_string, pattern=pattern, force_regex=force_regex
+		)
+
+	def check_enable_mode(self, check_string="#"):
+		"""Check if in enable mode. Return boolean.
+
+		:param check_string: Identification of privilege mode from device
+		:type check_string: str
+		"""
+		return super().check_enable_mode(
+			check_string=check_string
+		)
+
+	def cleanup(self):
+		"""Gracefully exit the SSH session."""
+		try:
+			self.exit_config_mode()
+		except Exception:
+			pass
+		# Always try to send final 'exit' regardless of whether exit_config_mode works or not.
+		self._session_log_fin = True
+		self.write_channel("exit" + self.RETURN)
+
+	def enable(
+		self,
+		cmd: str = "enable",
+		pattern: str = "ssword",
+		enable_pattern: str = "#",
+		re_flags: int = re.IGNORECASE,
+		) -> str:
+		"""Enter enable mode.
+
+		:param cmd: Device command to enter enable mode
+
+		:param pattern: pattern to search for indicating device is waiting for password
+
+		:param enable_pattern: pattern indicating you have entered enable mode
+
+		:param re_flags: Regular expression flags used in conjunction with pattern
+		"""
+		return super().enable(
+			cmd=cmd, pattern=pattern, enable_pattern=enable_pattern, re_flags=re_flags
+		)
+
+	def exit_config_mode(self, exit_config="exit", pattern="#"):
+		"""Exit from configuration mode.
+
+		:param exit_config: Command to exit configuration mode
+		:type exit_config: str
+
+		:param pattern: Pattern to terminate reading of channel
+		:type pattern: str
+		"""
+		return super().exit_config_mode(
+			exit_config=exit_config, pattern=pattern
+		)
+
+	def exit_enable_mode(self, exit_command="disable"):
+		"""Exit enable mode.
+
+		:param exit_command: Command that exits the session from privileged mode
+		:type exit_command: str
+		"""
+		return super().exit_enable_mode(
+			exit_command=exit_command
+		)
+
+	def send_config_set(
+		self,
+		config_commands = None,
+		exit_config_mode: bool = True,
+		read_timeout: float = None,
+		delay_factor: float = 1.0,
+		max_loops: int = 150,
+		strip_prompt: bool = False,
+		strip_command: bool = False,
+		config_mode_command: str = None,
+		cmd_verify: bool = True,
+		enter_config_mode: bool = False,
```

Comment:
Why is this `enter_config_mode=False`?

Reply:
Because there is no universal configuration mode on these devices.  I think we spoke about this before, but you can go into different config modes, such as:  config network, config data, config system, config voip, config troubleshooting.  Therefore, by default you would want that de-active so the user would choose the configuration mode they want to go into.  If you want to choose a default value, you could and the user could override the parameter, but not sure what that would serve.

Reply:
Yes, understood...but you always need to pick one of those i.e. it always needs to enter config mode (using some command)--correct?

We can just put a check that `config_mode_command is not None` and raise and exception if it is.

Reply:
That is correct, you'd have to choose a config option to go into the appropriate mode.  I typically just include the "config system" in my base script, otherwise it would fail to enter config mode anyways.